### PR TITLE
tt: added tt create module

### LINKED ISSUE #12: tt create
see `cartridge create`

### PR DESCRIPTION
**Pull request by psergee**
tt create allows a user to create tarantool application from a template.
tt create accepts template name as a parameter and searches for it in 
```yaml
  templates:
    - path:
    - path:
```
of current environment. Found template will be used for application creation.
'basic' is the default template name.
Supported flags:
```
FLAGS
  -f, --force              Force rewrite application directory if already exists
  -h, --help               help for create
      --name string        Application name
  -s, --non-interactive    Non-interactive mode
      --var stringArray    Variable definition. Usage: --var var_name=value
      --vars-file string   Variables definition file path
```
Invocation example:
`tt create --var name=name --name=app1 --non-interactive`

Closes #12

### CODE REVIEW

File: README.rst
Comment (vr009) [inline README.rst:181]: AFAIS, the run module has nothing to do with this PR, please move this change to a separate commit for having clear history. [2]

File: CHANGELOG.md
Comment (vr009) [inline CHANGELOG.md:27]: AFAIS, the run module has nothing to do with this PR, please move this change to a separate commit for having clear history. [1]

File: cli/config/config.go
Comment (vr009) [inline cli/config/config.go:74]: Missing some comment here.

File: cli/cmdcontext/cmdcontext.go
Comment (vr009) [inline cli/cmdcontext/cmdcontext.go:15]: Please, add a comment here.

**Review by vr009 — COMMENTED** [review]
The code looks good. It would be also nice to have some description how to use it for users, but I don'y know if we have some place to leave the instructions for this. I will leave more feedback for the code later.

**Comment by vr009** [comment]
It is also accepted (here) to mark a commit with `closes #N` instead of resolves.

File: cli/cmdcontext/cmdcontext.go
Comment (psergee) [reply cli/cmdcontext/cmdcontext.go]: Added

File: cli/config/config.go
Comment (psergee) [reply cli/config/config.go]: Done.

**Comment by psergee** [comment]
> The code looks good. It would be also nice to have some description how to use it for users, but I don'y know if we have some place to leave the instructions for this. I will leave more feedback for the code later.

Examples added to the help:
```
EXAMPLES
  
# Create an application from basic template.

    $ tt create

# Create app1 application from cartridge template, set user_name value,
# force replacing of application directory (app1) if it exists. User interaction is disabled.

    $ tt create cartridge --name=app1 --var user_name=admin -f --non-interactive
```

File: cli/cmd/create.go
Comment (0x501D) [inline cli/cmd/create.go:26]: The dot at the end of a sentence in the message is at odds with the general style of the rest help of the commands.

File: cli/create/create.go
Comment (0x501D) [inline cli/create/create.go:16]: This function never returns error, why not void?

File: cli/cmd/create.go
Comment (psergee) [reply cli/cmd/create.go:26]: Fixed.

File: cli/create/create.go
Comment (psergee) [reply cli/create/create.go:16]: Removed error return.

**Review by vr009 — APPROVED** [review]
LGTM
upd. after rebase :)

File: cli/util/util.go
Comment (LeonidVas) [inline cli/util/util.go:288]: GetMissedBinaries -> getMissedBinaries

**Review by LeonidVas — CHANGES_REQUESTED** [review]
Hi. I briefly reviewed the patch and plan to explore it further. The idea of using the "chain of responsibility" pattern looks interesting.
I have several comments:
* "Update cartridge-cli submodule" -> "cartridge cli: bump new version".
* It would be nice to add reasons to bump the new version to commit message. I think this was done not just for fun.

File: cli/create/create.go
Comment (vr009) [reply cli/create/create.go:16]: There is another opinion. See https://github.com/tarantool/tt/pull/119#discussion_r952587054.

File: CHANGELOG.md
Comment (psergee) [reply CHANGELOG.md:27]: This is not a typo, this is missing newline symbol. Since, I have added new line, this change is related to the patch. I will not write a not on the same line.

File: cli/create/create.go
Comment (psergee) [reply cli/create/create.go:16]: I don't like the idea of putting all command specific data into one God-structure. When I work with create command, I want to fill Create context, not cmd context, which contains Running, Connect, etc, data.

**Review by LeonidVas — COMMENTED** [review]
Hi! Thank you for the patchset.
Patchset already has two LGTM so I did not see it in detail, rather an introductory one.
I expect the following PR with a description of the functionality in the README and with examples in https://github.com/tarantool/tt/blob/master/doc/examples.rst

`Fix typos in code and comment` -> Commit title should be of a form `<subsystem>: <description>` (https://github.com/tarantool/tarantool/wiki/Code-review-procedure#commit-message)

This is a big and good work, and I believe that in the future you will split the functionality into several commits. Right now it's very difficult to review.

See comments bellow (some of them may be irrelevant because your update happened in the middle of my review.).

**Comment by LeonidVas** [comment]
Perhaps some of the questions are easier to discuss by voice. So ping if you need.

File: cli/cmd/create.go
Comment (psergee) [reply cli/cmd/create.go:35]: Because I do not see error messages from fatalf in my terminal.

File: cli/cmdcontext/cmdcontext.go
Comment (psergee) [reply cli/cmdcontext/cmdcontext.go:111]: These commands have access to ConfigPath common data. And they can make filepath.Dir if config directory is required. Do we really need to store file path and its directory next to each other?

File: cli/config/config.go
Comment (psergee) [reply cli/config/config.go:76]: There is no point in using a slice of pointers unless you need to express any elements as nil. Slice by itself behaves like a pointer, so this member looks fine for me.

File: cli/create/create.go
Comment (psergee) [reply cli/create/create.go]: > There is no restriction that a module can fill, use only one context.
There is no restrictions. But there must be.
I will separate contexts later :)

**Review by LeonidVas — APPROVED** [review]

**Pull request merged**
